rclcpp/include/rclcpp/strategies/static_memory_strategy.hpp

```diff
@@ -28,6 +28,12 @@ namespace memory_strategies
 namespace static_memory_strategy
 {
 
+template<size_t max_subscriptions_ = 10,
+size_t max_services_ = 10,
+size_t max_clients_ = 10,
+size_t max_executables_ = 1,
+size_t max_guard_conditions_ = 2,
+size_t pool_size_ = 1024>
```

Comment:
I think the constructor arguments are preferred since that make the class not a template and therefore implementable in a source file rather then in a header and therefore faster to compile.